src/lib/utils/concepts.h

```diff
@@ -43,6 +44,17 @@ static constexpr bool all_same_v = all_same<Ts...>::value;
 
 namespace detail {
 
+/**
+ * @brief Helper class to pass literal strings to C++ templates
+ */
+template <size_t N>
+class StringLiteral final {
+   public:
+      constexpr StringLiteral(const char (&str)[N]) { std::copy_n(str, N, value); }
+
+      char value[N];
+};
```

Comment:
I'm not quite happy with misusing the `concepts.h` for this. But I didn't find a better (public) utility header. `types.h` perhaps, but also not really. I didn't want to set up a new header just for this helper either.